### PR TITLE
PSR-1539 Added pre pop list page

### PR DESCRIPTION
**Pull request by westwater**
requires https://github.com/hmrc/pension-scheme-return-stub/pull/89

### CODE REVIEW

File: test/controllers/nonsipp/accountingperiod/AccountingPeriodListControllerSpec.scala [hunk 1]
```diff
@@ -73,7 +73,7 @@ class AccountingPeriodListControllerSpec extends ControllerBaseSpec {
 
       forAll(srnGen, rowsGen, modeGen) { (srn, rows, mode) =>
         val viewModel = AccountingPeriodListController.viewModel(srn, mode, rows)
-        viewModel.page.rows.length mustBe rows.length
+        viewModel.page.sections.length mustBe rows.length
```
[the remark this answers is not in the document]
Comment (westwater) [reply test/controllers/nonsipp/accountingperiod/AccountingPeriodListControllerSpec.scala:76]: haven't touched tests yet outside of renaming - will sort that out now 👍

File: app/controllers/PSRController.scala [hunk 2]
```diff
@@ -119,6 +121,14 @@ abstract class PSRController extends FrontendBaseController with I18nSupport {
         }
     }
 
+  implicit class EitherOps[A](maybe: Either[String, A]) {
+    def getOrRecoverJourney(implicit logger: Logger): Either[Result, A] =
+      maybe.leftMap { errMsg =>
+        logger.warn(errMsg)
```
[the remark this answers is not in the document]
Comment (westwater) [reply app/controllers/PSRController.scala:126]: yeah all errors are around missing pages

File: test/views/ListViewSpec.scala [hunk 3]
```diff
@@ -98,54 +98,55 @@ class ListViewSpec extends ViewSpec {
...
-          "current page is 2 (final page), there are 4 rows and page size is 3" in {
-            paginationTest(4, 2, 3) { html =>
-              summaryListRows(html).size mustEqual 1
-              val elems = paginationElements(html)
-              elems(2).isCurrentPage mustEqual true
-              elems.map(_.text) mustEqual List("Previous", "1", "2")
-            }
-          }
-        }
+        // TODO uncomment and fix when pagination is fixed for pre-pop list pages
```
Comment (kberec) [inline test/views/ListViewSpec.scala:101]: @westwater could you please mention this commented out section in the Jira ticket for pagination please

**Review by kberec — APPROVED** [review]

File: test/views/ListViewSpec.scala (hunk 3 above)
Comment (westwater) [reply test/views/ListViewSpec.scala:101]: added here https://jira.tools.tax.service.gov.uk/browse/PSR-1643

**Pull request merged**